client/v2/tx/tx.go

```diff
@@ -150,66 +160,30 @@ func SimulateTx(ctx client.Context, flagSet *pflag.FlagSet, msgs ...transaction.
 // BroadcastTx attempts to generate, sign and broadcast a transaction with the
 // given set of messages. It will also simulate gas requirements if necessary.
 // It will return an error upon failure.
-func BroadcastTx(clientCtx client.Context, txf Factory, broadcaster broadcast.Broadcaster, msgs ...transaction.Msg) error {
+func BroadcastTx(ctx context.Context, txf Factory, broadcaster broadcast.Broadcaster, msgs ...transaction.Msg) ([]byte, error) {
 	if txf.simulateAndExecute() {
 		err := txf.calculateGas(msgs...)
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	err := txf.BuildUnsignedTx(msgs...)
 	if err != nil {
-		return err
-	}
-
-	if !clientCtx.SkipConfirm {
-		encoder := txf.txConfig.TxJSONEncoder()
-		if encoder == nil {
-			return errors.New("failed to encode transaction: tx json encoder is nil")
-		}
-
-		unsigTx, err := txf.getTx()
-		if err != nil {
-			return err
-		}
-		txBytes, err := encoder(unsigTx)
-		if err != nil {
-			return fmt.Errorf("failed to encode transaction: %w", err)
-		}
-
-		if err := clientCtx.PrintRaw(txBytes); err != nil {
-			_, _ = fmt.Fprintf(os.Stderr, "error: %v\n%s\n", err, txBytes)
-		}
-
-		buf := bufio.NewReader(os.Stdin)
-		ok, err := input.GetConfirmation("confirm transaction before signing and broadcasting", buf, os.Stderr)
-		if err != nil {
-			_, _ = fmt.Fprintf(os.Stderr, "error: %v\ncanceled transaction\n", err)
-			return err
-		}
-		if !ok {
-			_, _ = fmt.Fprintln(os.Stderr, "canceled transaction")
-			return nil
-		}
+		return nil, err
 	}
 
-	signedTx, err := txf.sign(clientCtx.CmdContext, true)
+	signedTx, err := txf.sign(ctx, true)
```

Comment:
This funcition still need to be split in two, otherwise we loose the ability to confirm